### ISSUE TITLE
On Mac: the extraneous .DS_Store file and .sync-conflict-...

### ISSUE THREAD

**Issue by donkoller** [kalabox/kalabox#196]
After kbox up, and then kbox start. (Not sure whether they were there before I last shut down my Mac.)
I see these showing up in git status:
        .sync-conflict-20150605-152513.DS_Store
	sites/.sync-conflict-20150605-152513.DS_Store
	sites/all/.sync-conflict-20150605-112527.DS_Store

I've added *.DS_Store to .gitignore.

**Comment by reynoldsalec** [comment]
Hey @donkoller, that's interesting, haven't seen that before. DS_Store files are used in Mac OSX for certain pieces of data about folders (mostly whether it has a custom icon or not). Sync-thing would have generated those .sync-conflict files, adding *.DS_Store to the sync-thing ignores in kalabox.json might be the best thing to do on all our default apps.

@bcauldwell or @pirog any opinions?

**Comment by pirog** [comment]
@donkoller i think @reynoldsalec is probably right here. You might also want to check to see if it is in your .gitignore file just for due diligence. 

I would try to add those to the shareIgnores in your kalabox.json for your app and see if that prevents this form happening again.

```
  "shareIgnores": [
    "sites/default/files/*",
    ".git",
    ".DS_Store"
  ],
```

If that works maybe we should consider adding this as a default?

**Comment by andrewmallis** [comment]
I'd assume Windows could exhibit similar behavior for 
Thumbs.db

**Comment by donkoller** [comment]
Mike, I added your shareIgnores suggestion to my kbox.json in the web app root.
Will post whether this recurs or not.

The .gitignore file contains:
# Ignore configuration files that may contain sensitive information.
sites/*/settings*.php

# Ignore paths that contain user-generated content.
sites/*/files
sites/*/private

.stfolder
.stignore

# Ignore Mac generated .DS_Store files and syncthing conflicts
*.DS_Store
.*.DS_Store

So, .gitignore is good.

 	
Don Koller | @dkrockville
BEA Enterprises, Inc.
17 Lake Potomac Ct. | Potomac, MD 20854 USA
office: 301-299-7900 | mobile 301-520-9820

web: bea-enterprises.com <http://bea-enterprises.com/> | email: don@bea-enterprises.com <mailto:don@bea-enterprises.com>

**Comment by donkoller** [comment]
After making this change to kbox.json, and running kbox up and kbox start, these files appeared in /code: 

and these in /code/sites:


However, git status does not show them! I think .gitignore is working.
Is there a way to make syncthing not sync certain files (.syncthingignore)?

I plan to do a Windows test soon; will I also need to add thumbs.db for that?

Don

**Comment by reynoldsalec** [comment]
@donkoller for some reason we can't see the file lists you pasted into your last comment, so I'm not sure exactly what's still syncing over, but adding those files (or, ideally, a pattern that eliminates them) to kalabox.json's shareIgnores array as you did for the other files should do the trick.

If you get a chance, could you try copying those file lists over again? Interested to see what's popping up so we can see if it'll be a problem for other people. Thanks!

**Comment by donkoller** [comment]
Filenames are:
In /code:
.sync-conflict-20150605-152513.sync-conflict-20150609-160751.DS_Store
.sync-conflict-20150606-135611.sync-conflict-20150609-160751.DS_Store

In /code/sites:
.sync-conflict-20150605-152513.sync-conflict-20150609-160751.DS_Store
.sync-conflict-20150606-135611.sync-conflict-20150609-160751.DS_Store

In case you want to look at the actual files, go here (where I’ve pre-pended code or code.sites to the file name):
https://www.dropbox.com/sh/86c6x5aatr6ctyl/AADeqMNggqgtJcMaaHsQSY8va?dl=0 


Is it OK to add a pattern in shareIgnores, like “.sync-conflict*”?

Regards,
Don

**Comment by pirog** [comment]
I think our sharesIgnore key in the kalabox.json (which ends up in the .stignore file) takes the same kind of wildcarding patterns as teh gitignore so you probably actually would want to do either `.sync-conflict*` as you suggested or `*.DS_Store` which might be better. 

Let let us know how it goes on Windows. 

I think we might want to just add a bunch of additional things like this to our default shareIgnores? Any thoughts here?

good generic candidates for most kinds of apps: 

```
# Packages #
############
*.7z
*.dmg
*.gz
*.bz2
*.iso
*.jar
*.rar
*.tar
*.zip
*.tgz

# OS generated files #
######################
.DS_Store*
ehthumbs.db
Icon

Thumbs.db
._*

# Vim generated files #
######################
*.un~

# SASS #
##########
.sass-cache
```

**Comment by donkoller** [comment]
Updated kbox.json:

  "shareIgnores": [
    "sites/default/files/*",
    ".git",
    ".DS_Store",
    ".sync_conflict*",
    "sites/.sync-conflict*"
  ],

Did kbox stop. kbox rebuild. kbox start.
Syncthing is still creating the .sync-conflict files.

Don

**Comment by pirog** [comment]
Hmmm. 

Can you actually post the steps that we can take to replicate these sync-conflict files showing up. 

If what was recommended above isn't working for you maybe we have a bug on our end to find.

**Comment by andrewmallis** [comment]
ehthumbs.db

only appear in Windows XP Media Center Edition
http://en.wikipedia.org/wiki/Windows_thumbnail_cache#Thumbs.db

suggest excluding lest we send the wrong message for platform support

**Comment by pirog** [comment]
good to know but jury is still out on XP support. It very well might work we've just had 0 trials so far.

**Comment by donkoller** [comment]
Modified the kbox.json to 

  "shareIgnores": [
    "sites/default/files/*",
    ".git",
    ".DS_Store",
    ".*DS_Store",
    "sites/.*DS_Store"
  ],

After kbox down, kbox up and kbox start, those .sync-conflict* files are no longer showing up in /code nor in /code/sites. It’s a fix!

By the way, I’d like to see those other generic files also ignored. In 0.8?

Regards,
Don

**Comment by pirog** [comment]
oh awesome!

yeah i think we are def going to add a bunch of these entries into 0.8.0. This will be a great improvement for all. THanks @donkoller

**Comment by donkoller** [comment]
One more thing …

Those sync-conflicts started showing up elsewhere. Not sure what triggers it.
Here is my new, improved ignore code in kbox.json:

  "shareIgnores": [
    "sites/default/files/*",
    ".git",
    ".DS_Store",
    ".*DS_Store",
    "sites/.*DS_Store"
    "sites/all/.*DS_Store"
    "sites/all/libraries/.*DS_Store"
    "sites/all/modules/.*DS_Store"
    "sites/all/themes/.*DS_Store"
  ],

The .sync-conflicts started showing up in sites/all/, but I added the others just to be sure.
Any other advice?

Regards,
Don

**Comment by donkoller** [comment]
Added missing commas…

  "shareIgnores": [
    "sites/default/files/*",
    ".git",
    ".DS_Store",
    ".*DS_Store",
    "sites/.*DS_Store”,
    "sites/all/.*DS_Store”,
    "sites/all/libraries/.*DS_Store”,
    "sites/all/modules/.*DS_Store”,
    "sites/all/themes/.*DS_Store"
  ],

**Comment by pirog** [comment]
nice. thanks don! this stuff should be in the new version

**Comment by bcauldwell** [comment]
Fixed for 0.8 aka cyrus-wreckingBall.

See commit. https://github.com/kalabox/kalabox/commit/3d8dad0e80ac08a808620737bb2c7d487a83bb3d
